[BUG] gio-unix: DesktopAppInfo should not be available on macos

**Bug description**

Regression in gtk-rs 0.22

GDesktopAppInfo should not be available on macos: in https://gitlab.gnome.org/GNOME/glib/-/blob/main/gio/meson.build?ref_type=heads#L420 the source file is only compiled when `glib_have_cocoa` is false.

In gtk-rs 0.21, `DesktopAppInfo` had a `cfg_condition` that is no longer present in 0.22:
- 0.21: https://github.com/gtk-rs/gtk-rs-core/blob/0.21.5/gio/Gir.toml#L766
- 0.22: https://github.com/gtk-rs/gtk-rs-core/blob/0.22.5/gio-unix/Gir.toml#L49

Trying to build on macos a program that uses `gio_unix::DesktopAppInfo` now gives a cryptic linker error:

```
          Undefined symbols for architecture arm64:
            "_g_desktop_app_info_get_type", referenced from:
                _$LT$gio_unix..auto..desktop_app_info..DesktopAppInfo$u20$as$u20$glib..types..StaticType$GT$::static_type::h56f1ea7f4f8aa2a2 in libgio_unix-61787655b8f4018c.rlib[3](gio_unix-61787655b8f4018c.gio_unix.4e1cd38a89ae7b9f-cgu.0.rcgu.o)
```

instead of the expected clear compilation error:

```
error[E0433]: failed to resolve: could not find `DesktopAppInfo` in `gio`
 --> src/lib.rs:2:19
  |
2 |     let _n = gio::DesktopAppInfo::new("foo.desktop");
  |                   ^^^^^^^^^^^^^^ could not find `DesktopAppInfo` in `gio`
```

--- a/gio-unix/Gir.toml
+++ b/gio-unix/Gir.toml
@@ -49,6 +49,7 @@ status = "generate"
 name = "GioUnix.DesktopAppInfo"
 status = "generate"
 manual_traits = ["DesktopAppInfoExtManual"]
+cfg_condition = "not(target_os = \"macos\")"
     [[object.function]]
     name = "get_boolean"
     # Retrieves the boolean value of a key

--- a/gio-unix/src/auto/mod.rs
+++ b/gio-unix/src/auto/mod.rs
@@ -2,7 +2,11 @@
 // from gir-files (https://github.com/gtk-rs/gir-files)
 // DO NOT EDIT
 
+#[cfg(not(target_os = "macos"))]
+#[cfg_attr(docsrs, doc(cfg(not(target_os = "macos"))))]
 mod desktop_app_info;
+#[cfg(not(target_os = "macos"))]
+#[cfg_attr(docsrs, doc(cfg(not(target_os = "macos"))))]
 pub use self::desktop_app_info::DesktopAppInfo;
 
 mod fd_message;

--- a/gio-unix/src/lib.rs
+++ b/gio-unix/src/lib.rs
@@ -10,7 +10,7 @@ pub use gio_unix_sys as ffi;
 mod auto;
 pub use auto::*;
 
-#[cfg(feature = "v2_58")]
+#[cfg(all(feature = "v2_58", not(target_os = "macos")))]
 mod desktop_app_info;
 mod fd_message;
 mod file_descriptor_based;
@@ -29,7 +29,7 @@ pub mod prelude {
 
     pub use super::auto::traits::*;
 
-    #[cfg(feature = "v2_58")]
+    #[cfg(all(feature = "v2_58", not(target_os = "macos")))]
     pub use crate::desktop_app_info::DesktopAppInfoExtManual;
 
     pub use crate::fd_message::FDMessageExtManual;

